server/src/inmantals/server.py

```diff
@@ -64,9 +65,149 @@ def __init__(self, message: str) -> None:
         self.message = message
 
 
+class Folder:
+    """
+    Wrapper class around a folder (inmanta module or inmanta project). This folder can either be standalone or live in a
+    workspace alongside other folders.
+
+    The constructor should not be called explicitly. Instantiation is done through the unpack_workspaces method
+    :param folder_uri: uri of the folder that is assumed to live in a workspace
+    :param name: name of the folder
+    :param handler: reference to the InmantaLSHandler responsible for this folder
+    """
+
+    folder_uri: URI
+    name: str
+    inmanta_project_dir: Union[Optional[tempfile.TemporaryDirectory], str]
+    handler: "InmantaLSHandler"
+
+    def __init__(self, folder_uri: URI, name: str, handler: "InmantaLSHandler"):
+        self.folder_uri = folder_uri
+        self.name = name
+        self.handler = handler  # Keep a reference to the handler for cleanup
```

Comment:
Am I right to think it is with this you fixed the issue I reported yesterday?

Reply:
Yes, the cleanup wasn't done right:
The create_tmp_project method used be part of the InmantaLSHandler class and create self.project_dir. The problem was that I moved this method to the Folder class, but I didn't change the cleanup logic(https://github.com/inmanta/vscode-inmanta/blob/662678c99471a8144c52f5a7e56e9aff8a1cb685/server/src/inmantals/server.py#L456-L458)